src/renderer/families/stellar/MemoTypeField.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import React, { useCallback, useState, useEffect } from "react";
+import { getAccountBridge } from "@ledgerhq/live-common/lib/bridge";
+import { StellarMemoType } from "@ledgerhq/live-common/lib/families/stellar/types";
+import Select from "~/renderer/components/Select";
+import invariant from "invariant";
+import type { Account, Transaction } from "@ledgerhq/live-common/lib/types";
+
+const MemoTypeField = ({
+  onChange,
+  account,
+  transaction,
+}: {
+  onChange: string => void,
+  account: Account,
+  transaction: Transaction,
+}) => {
+  invariant(transaction.family === "stellar", "MemoTypeField: stellar family expected");
+
+  const bridge = getAccountBridge(account);
+
+  const options = StellarMemoType.map(type => ({
+    label: type,
+    value: type,
+  }));
+
+  const [selectedMemoType, setSelectedMemoType] = useState(options[0]);
+
+  const onMemoTypeChange = useCallback(
+    memoType => {
+      setSelectedMemoType(memoType);
+      onChange(bridge.updateTransaction(transaction, { memoType: memoType.value }));
+    },
+    [onChange, bridge, transaction],
+  );
+
+  useEffect(() => {
+    setSelectedMemoType(options.find(option => option.value === transaction.memoType));
+  }, [transaction.memoType, options]);
```

Comment:
mmh, this makes me thing that `selectedMemoType` should not be a state but it should be a variable. when you have a state that depends on another state but essentially don't need that state to evolve BUT to follow the first state, it is not a state, it should be a "derived data" calculated / memoized from the initial state. 

would it work if we would:
- remove line `const [selectedMemoType, setSelectedMemoType] = useState(options[0]);`
- replace it with just: `const selectedMemoType = options.find(option => option.value === transaction.memoType) || options[0]`
- remove completely this effect